### PR TITLE
TF2: add hyperopt_report vis api and related unit tests

### PR DESCRIPTION
**Pull request by jimthompson5802**
# Code Pull Requests

Added vis api for `hyperopt_report()` and related unit test
Added unit test for `hyperopt.run.hyperopt()`

### CODE REVIEW

File: tests/integration_tests/test_hyperopt.py [hunk 1]
```diff
@@ -126,4 +128,69 @@ def test_hyperopt_executor_with_metric(csv_filename):
...
+                'high': 5,
+                'space': 'linear',
+                'steps': 4
+            }
+        },
+        "goal": "minimize",
+        'output_feature': output_feature_name,
+        'validation_metrics': 'loss',
+        'executor': {'type': 'serial'},
+        'sampler': {'type': 'random', 'num_samples': 5}
```
Comment (w4nderlust) [inline tests/integration_tests/test_hyperopt.py:176]: To keep this fast, 2 samples is sufficient, we are already testing more samples with the other hyperopt tests

File: tests/integration_tests/test_visualization_api.py [hunk 2]
```diff
@@ -817,3 +819,75 @@ def test_frequency_vs_f1_vis_api(csv_filename):
...
+                'high': 5,
+                'space': 'linear',
+                'steps': 4
+            }
+        },
+        "goal": "minimize",
+        'output_feature': output_feature_name,
+        'validation_metrics': 'loss',
+        'executor': {'type': 'serial'},
+        'sampler': {'type': 'random', 'num_samples': 5}
```
Comment (w4nderlust) [inline tests/integration_tests/test_visualization_api.py:868]: Same here, 2 samples is sufficient

File: tests/integration_tests/test_hyperopt.py (hunk 1 above)
Comment (jimthompson5802) [reply tests/integration_tests/test_hyperopt.py]: done.

File: tests/integration_tests/test_visualization_api.py (hunk 2 above)
Comment (jimthompson5802) [reply tests/integration_tests/test_visualization_api.py]: done

**Comment by jimthompson5802** [comment]
Added `hyperopt_report` and `hyperopt_hiplot` visualization apis and related unit test.
Note: for the unit tests, separated out the `hyperopt.run.hyperopt` into a module level fixture.  This runs supports running `hyperopt.run.hyperopt` only once and have the results usable in the two visualization tests.  Helps minimize run-time.  I think we should be able to do something similar with the other visualization tests to reduce their run-times as well.

Local run of the visualization_api unit tests.  The hyperopt visualization api tests at the end
```
====================================================== test session starts ======================================================
platform linux -- Python 3.6.9, pytest-6.0.2, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /opt/project
plugins: forked-1.3.0, pycharm-0.7.0, xdist-2.1.0, typeguard-2.9.1
collected 24 items

../../tests/integration_tests/test_visualization_api.py::test_learning_curves_vis_api PASSED                              [  4%]
../../tests/integration_tests/test_visualization_api.py::test_compare_performance_vis_api PASSED                          [  8%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifier_performance_from_prob_vis_api PASSED     [ 12%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifier_performance_from_pred_vis_api PASSED     [ 16%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifiers_performance_subset_vis_api PASSED       [ 20%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifiers_performance_changing_k_vis_api PASSED   [ 25%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifiers_multiclass_multimetric_vis_api PASSED   [ 29%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifiers_predictions_vis_api PASSED              [ 33%]
../../tests/integration_tests/test_visualization_api.py::test_compare_classifiers_predictions_distribution_vis_api PASSED [ 37%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_vis_api PASSED                      [ 41%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_data_vs_acc_vis_api PASSED          [ 45%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_data_vs_acc_subset_vis_api PASSED   [ 50%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_data_vs_acc_subset_per_class_vis_api PASSED [ 54%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_2thresholds_2d_vis_api PASSED       [ 58%]
../../tests/integration_tests/test_visualization_api.py::test_confidence_thresholding_2thresholds_3d_vis_api PASSED       [ 62%]
../../tests/integration_tests/test_visualization_api.py::test_binary_threshold_vs_metric_vis_api PASSED                   [ 66%]
../../tests/integration_tests/test_visualization_api.py::test_roc_curves_vis_api PASSED                                   [ 70%]
../../tests/integration_tests/test_visualization_api.py::test_roc_curves_from_test_statistics_vis_api PASSED              [ 75%]
../../tests/integration_tests/test_visualization_api.py::test_calibration_1_vs_all_vis_api PASSED                         [ 79%]
../../tests/integration_tests/test_visualization_api.py::test_calibration_multiclass_vis_api PASSED                       [ 83%]
../../tests/integration_tests/test_visualization_api.py::test_confusion_matrix_vis_api PASSED                             [ 87%]
../../tests/integration_tests/test_visualization_api.py::test_frequency_vs_f1_vis_api PASSED                              [ 91%]
../../tests/integration_tests/test_visualization_api.py::test_hyperopt_report_vis_api PASSED                              [ 95%]
../../tests/integration_tests/test_visualization_api.py::test_hyperopt_hiplot_vis_api PASSED                              [100%]

```

**Comment by w4nderlust** [comment]
The fixture idea is very good, in the future we can consider applying it in other places.

**Pull request merged**